File: system-test/sec-tests/javascript/sec_27_session_management.js
// SEC-27: Session Management Defects
// Test: TC-CR-033
const express = require('express');
const session = require('express-session');
const app = express();

app.use(session({
  secret: 'keyboard cat',  // Weak secret
  resave: true,
  saveUninitialized: true,
  cookie: {
    secure: false,  // Missing Secure flag
    httpOnly: false,  // Missing HttpOnly flag
    sameSite: 'none'  // Permissive SameSite
  }
}));

File: system-test/sec-tests/java/Sec27SessionManagement.java
// SEC-27: Session Management Defects
// Test: TC-CR-033
package com.example;

import javax.servlet.http.HttpSession;

public class SessionService {

    public void configureSession(HttpSession session) {
        // Missing security flags
        session.setMaxInactiveInterval(86400);  // 24h timeout — too long
    }
}
